servers/rendering/renderer_scene_render.h

```diff
@@ -43,38 +44,119 @@ class RendererSceneRender {
-	virtual void geometry_instance_set_parent_fade_alpha(GeometryInstance *p_geometry_instance, float p_alpha) = 0;
-	virtual void geometry_instance_set_use_baked_light(GeometryInstance *p_geometry_instance, bool p_enable) = 0;
-	virtual void geometry_instance_set_use_dynamic_gi(GeometryInstance *p_geometry_instance, bool p_enable) = 0;
-	virtual void geometry_instance_set_use_lightmap(GeometryInstance *p_geometry_instance, RID p_lightmap_instance, const Rect2 &p_lightmap_uv_scale, int p_lightmap_slice_index) = 0;
-	virtual void geometry_instance_set_lightmap_capture(GeometryInstance *p_geometry_instance, const Color *p_sh9) = 0;
-	virtual void geometry_instance_set_instance_shader_parameters_offset(GeometryInstance *p_geometry_instance, int32_t p_offset) = 0;
-	virtual void geometry_instance_set_cast_double_sided_shadows(GeometryInstance *p_geometry_instance, bool p_enable) = 0;
-
-	virtual uint32_t geometry_instance_get_pair_mask() = 0;
-	virtual void geometry_instance_pair_light_instances(GeometryInstance *p_geometry_instance, const RID *p_light_instances, uint32_t p_light_instance_count) = 0;
-	virtual void geometry_instance_pair_reflection_probe_instances(GeometryInstance *p_geometry_instance, const RID *p_reflection_probe_instances, uint32_t p_reflection_probe_instance_count) = 0;
-	virtual void geometry_instance_pair_decal_instances(GeometryInstance *p_geometry_instance, const RID *p_decal_instances, uint32_t p_decal_instance_count) = 0;
-	virtual void geometry_instance_pair_voxel_gi_instances(GeometryInstance *p_geometry_instance, const RID *p_voxel_gi_instances, uint32_t p_voxel_gi_instance_count) = 0;
+		virtual void _mark_dirty() = 0;
+
+		virtual void set_skeleton(RID p_skeleton) = 0;
+		virtual void set_material_override(RID p_override) = 0;
+		virtual void set_material_overlay(RID p_overlay) = 0;
+		virtual void set_surface_materials(const Vector<RID> &p_materials) = 0;
+		virtual void set_mesh_instance(RID p_mesh_instance) = 0;
+		virtual void set_transform(const Transform3D &p_transform, const AABB &p_aabb, const AABB &p_transformed_aabb) = 0;
+		virtual void set_lod_bias(float p_lod_bias) = 0;
+		virtual void set_layer_mask(uint32_t p_layer_mask) = 0;
+		virtual void set_fade_range(bool p_enable_near, float p_near_begin, float p_near_end, bool p_enable_far, float p_far_begin, float p_far_end) = 0;
+		virtual void set_parent_fade_alpha(float p_alpha) = 0;
+		virtual void set_transparency(float p_transparency) = 0;
+		virtual void set_use_baked_light(bool p_enable) = 0;
+		virtual void set_use_dynamic_gi(bool p_enable) = 0;
+		virtual void set_use_lightmap(RID p_lightmap_instance, const Rect2 &p_lightmap_uv_scale, int p_lightmap_slice_index) = 0;
+		virtual void set_lightmap_capture(const Color *p_sh9) = 0;
+		virtual void set_instance_shader_parameters_offset(int32_t p_offset) = 0;
+		virtual void set_cast_double_sided_shadows(bool p_enable) = 0;
+
+		virtual Transform3D get_transform() = 0;
```

Comment:
I've added get_transform and get_aabb here even though they were only defined in the RD renderer. They're being called on the base pointer.

I could change it and introduce a GeometryInstanceRD class inbetween and do a static cast where they are called but it feels like overkill. Happy to change it though.

@reduz what do you think?

Reply:
I think it is fine here. We will likely end up needing them in GLES3 once we add VoxelGI anyway.

Reply:
Yup, and it decreases the likelihood that someone implements another render device based renderer and inherits from the wrong class and misses out on these functions. The remaining question here is whether we should add more access methods or if that is overkill.